src/test/java/com/powsybl/openloadflow/ac/AcLoadFlowWithCachingTest.java

```diff
@@ -0,0 +1,141 @@
+
+/**
+ * @author Geoffroy Jamgotchian <geoffroy.jamgotchian at rte-france.com>
+ */
+class AcLoadFlowWithCachingTest {
+
+    private LoadFlow.Runner loadFlowRunner;
+
+    private LoadFlowParameters parameters;
+
+    @BeforeEach
+    void setUp() {
+        loadFlowRunner = new LoadFlow.Runner(new OpenLoadFlowProvider(new DenseMatrixFactory()));
+        parameters = new LoadFlowParameters();
+        OpenLoadFlowParameters.create(parameters)
+                .setNetworkCacheEnabled(true);
+        NetworkCache.INSTANCE.clear();
+    }
+
+    @Test
+    void testLoadP() {
+        var network = EurostagFactory.fix(EurostagTutorialExample1Factory.create());
+        var load = network.getLoad("LOAD");
+        var gen = network.getGenerator("GEN");
+
+        var result = loadFlowRunner.run(network, parameters);
+        assertEquals(LoadFlowResult.ComponentResult.Status.CONVERGED, result.getComponentResults().get(0).getStatus());
+        assertEquals(4, result.getComponentResults().get(0).getIterationCount());
+        assertActivePowerEquals(600, load.getTerminal());
+        assertActivePowerEquals(-605.559, gen.getTerminal());
+
+        load.setP0(620);
+
+        result = loadFlowRunner.run(network, parameters);
+        assertEquals(LoadFlowResult.ComponentResult.Status.CONVERGED, result.getComponentResults().get(0).getStatus());
+        assertEquals(3, result.getComponentResults().get(0).getIterationCount());
+        assertActivePowerEquals(620, load.getTerminal());
+        assertActivePowerEquals(-625.895, gen.getTerminal());
+    }
+
+    @Test
+    void testLoadPAndSlackDistributionOnLoads() {
+        var network = DistributedSlackNetworkFactory.createNetworkWithLoads();
+        var l14 = network.getBranch("l14");
+        var l24 = network.getBranch("l24");
+        var l34 = network.getBranch("l34");
+        var l2 = network.getLoad("l2");
+        var l4 = network.getLoad("l4");
+
+        parameters.setBalanceType(LoadFlowParameters.BalanceType.PROPORTIONAL_TO_LOAD);
+        var result = loadFlowRunner.run(network, parameters);
+        assertEquals(LoadFlowResult.ComponentResult.Status.CONVERGED, result.getComponentResults().get(0).getStatus());
+        assertEquals(3, result.getComponentResults().get(0).getIterationCount());
+        assertActivePowerEquals(70.588, l2.getTerminal());
+        assertActivePowerEquals(164.706, l4.getTerminal());
+        assertActivePowerEquals(64.706, l14.getTerminal1());
+        assertActivePowerEquals(129.412, l24.getTerminal1());
+        assertActivePowerEquals(-58.824, l34.getTerminal1());
+
+        l2.setP0(75);
+        l4.setP0(125);
+        result = loadFlowRunner.run(network, parameters);
+        assertEquals(LoadFlowResult.ComponentResult.Status.CONVERGED, result.getComponentResults().get(0).getStatus());
+        assertEquals(1, result.getComponentResults().get(0).getIterationCount());
+        assertActivePowerEquals(75.0, l2.getTerminal()); // FIXME, should be 88.235
```

Comment:
@annetill this is because LfAggregatedLoadsImpl is not updated (absVariableLoadTargetP for load balance) and this is why I tried to fix in closed PR https://github.com/powsybl/powsybl-open-loadflow/pull/640

Reply:
feature discarded from PR